File: AllSamples/Samples/Samples_Source/peer networking/peeridentity viewer/vb/dlgnewidentity.vb
'=====================================================================
'  File:      DlgNewIdentityInfo.vb
'
'  Summary:   Dialog to create a new PeerIdentity.
'
'---------------------------------------------------------------------
' This file is part of the Microsoft .NET Framework SDK Code Samples.
' 
' Copyright (C) Microsoft Corporation.  All rights reserved.
' 
' This source code is intended only as a supplement to Microsoft
' Development Tools and/or on-line documentation.  See these other
' materials for detailed information regarding Microsoft code samples.
' 
' THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
' KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
' IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
' PARTICULAR PURPOSE.
'---------------------------------------------------------------------

Option Explicit On
Option Strict On

Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports System.Net.PeerToPeer
Imports System.Windows.Forms


Public Class DlgNewIdentity
    Inherits System.Windows.Forms.Form

    Private identityValue As PeerIdentity = Nothing

    Private textBoxName As System.Windows.Forms.TextBox
    Private textBoxClassifier As System.Windows.Forms.TextBox
    Private WithEvents buttonOk As System.Windows.Forms.Button
    Private WithEvents buttonCancel As System.Windows.Forms.Button
    Private WithEvents labelName As System.Windows.Forms.Label
    Private WithEvents labelClassifier As System.Windows.Forms.Label

    Private components As System.ComponentModel.Container = Nothing

    ' Constructor to create the dialog.
    '
    Public Sub New()
        InitializeComponent()

        ' Default the name to the user's login name.
        textBoxName.Text = System.Environment.UserName

    End Sub 'New

    ' Clean up any resources being used.
    '
    Protected Overrides Sub Dispose(ByVal disposing As Boolean)
        If disposing Then
            If Not (components Is Nothing) Then
                components.Dispose()
            End If
        End If
        MyBase.Dispose(disposing)

    End Sub 'Dispose

#Region "Windows Form Designer generated code"

    ' <summary>
    ' Required method for Designer support - do not modify
    ' the contents of this method with the code editor.
    ' </summary>
    Private Sub InitializeComponent()
        Me.labelName = New System.Windows.Forms.Label
        Me.textBoxName = New System.Windows.Forms.TextBox
        Me.textBoxClassifier = New System.Windows.Forms.TextBox
        Me.labelClassifier = New System.Windows.Forms.Label
        Me.buttonOk = New System.Windows.Forms.Button
        Me.buttonCancel = New System.Windows.Forms.Button
        Me.SuspendLayout()
        '
        'labelName
        '
        Me.labelName.Location = New System.Drawing.Point(16, 16)
        Me.labelName.Name = "labelName"
        Me.labelName.Size = New System.Drawing.Size(48, 16)
        Me.labelName.TabIndex = 0
        Me.labelName.Text = "Name:"
        '
        'textBoxName
        '
        Me.textBoxName.Anchor = CType(((System.Windows.Forms.AnchorStyles.Top Or System.Windows.Forms.AnchorStyles.Left) _
                    Or System.Windows.Forms.AnchorStyles.Right), System.Windows.Forms.AnchorStyles)
        Me.textBoxName.Location = New System.Drawing.Point(72, 16)
        Me.textBoxName.Name = "textBoxName"
        Me.textBoxName.Size = New System.Drawing.Size(204, 19)
        Me.textBoxName.TabIndex = 1
        '
        'textBoxClassifier
        '
        Me.textBoxClassifier.Anchor = CType(((System.Windows.Forms.AnchorStyles.Top Or System.Windows.Forms.AnchorStyles.Left) _
                    Or System.Windows.Forms.AnchorStyles.Right), System.Windows.Forms.AnchorStyles)
        Me.textBoxClassifier.Location = New System.Drawing.Point(72, 56)
        Me.textBoxClassifier.Name = "textBoxClassifier"
        Me.textBoxClassifier.Size = New System.Drawing.Size(204, 19)
        Me.textBoxClassifier.TabIndex = 3
        '
        'labelClassifier
        '
        Me.labelClassifier.Location = New System.Drawing.Point(16, 56)
        Me.labelClassifier.Name = "labelClassifier"
        Me.labelClassifier.Size = New System.Drawing.Size(56, 16)
        Me.labelClassifier.TabIndex = 2
        Me.labelClassifier.Text = "Classifier:"
        '
        'buttonOk
        '
        Me.buttonOk.Anchor = CType((System.Windows.Forms.AnchorStyles.Bottom Or System.Windows.Forms.AnchorStyles.Right), System.Windows.Forms.AnchorStyles)
        Me.buttonOk.Location = New System.Drawing.Point(71, 96)
        Me.buttonOk.Name = "buttonOk"
        Me.buttonOk.TabIndex = 4
        Me.buttonOk.Text = "OK"
        '
        'buttonCancel
        '
        Me.buttonCancel.Anchor = CType((System.Windows.Forms.AnchorStyles.Bottom Or System.Windows.Forms.AnchorStyles.Right), System.Windows.Forms.AnchorStyles)
        Me.buttonCancel.DialogResult = System.Windows.Forms.DialogResult.Cancel
        Me.buttonCancel.Location = New System.Drawing.Point(159, 96)
        Me.buttonCancel.Name = "buttonCancel"
        Me.buttonCancel.TabIndex = 5
        Me.buttonCancel.Text = "Cancel"
        '
        'DlgNewIdentity
        '
        Me.AcceptButton = Me.buttonOk
        Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
        Me.CancelButton = Me.buttonCancel
        Me.ClientSize = New System.Drawing.Size(292, 133)
        Me.ControlBox = False
        Me.Controls.Add(Me.buttonCancel)
        Me.Controls.Add(Me.buttonOk)
        Me.Controls.Add(Me.textBoxName)
        Me.Controls.Add(Me.labelName)
        Me.Controls.Add(Me.textBoxClassifier)
        Me.Controls.Add(Me.labelClassifier)
        Me.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedDialog
        Me.Name = "DlgNewIdentity"
        Me.StartPosition = System.Windows.Forms.FormStartPosition.CenterParent
        Me.Text = "Create New Identity"
        Me.ResumeLayout(False)
        Me.PerformLayout()

    End Sub 'InitializeComponent
#End Region

    ' The PeerIdentity that was imported.
    '
    Public ReadOnly Property Identity() As PeerIdentity
        Get
            Return identityValue
        End Get
    End Property

    ' Validate the data in the dialog and create the Identity.
    '
    Private Sub buttonOk_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles buttonOk.Click
        If CreateIdentity() Then
            Me.DialogResult = DialogResult.OK
            Me.Close()
        End If

    End Sub 'buttonOk_Click

    ' Try to create the identity.
    '
    Private Function CreateIdentity() As Boolean
        ' Validate the classifier
        textBoxClassifier.Text = textBoxClassifier.Text.Trim()
        If textBoxClassifier.TextLength >= 150 Then
            MessageBox.Show(Me, "The classifier must be less than 150 characters", "Invalid Classifier", MessageBoxButtons.OK, MessageBoxIcon.Error)
            Return False
        End If

        ' Validate the friendly name
        textBoxName.Text = textBoxName.Text.Trim()
        If textBoxName.TextLength >= 256 Then
            MessageBox.Show(Me, "Identity friendly must be less than 256 characters", "Invalid Name", MessageBoxButtons.OK, MessageBoxIcon.Error)
            Return False
        End If

        Try
            identityValue = New PeerIdentity(textBoxName.Text, textBoxClassifier.Text)
            Return True
        Catch e As Exception
            Utilities.DisplayException(e, Me)
            Return False
        End Try

    End Function 'CreateIdentity

    ' Close the dialog without creating a new identity.
    '
    Private Sub buttonCancel_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles buttonCancel.Click
        Me.Close()
    End Sub 'buttonCancel_Click

End Class 'DlgNewIdentity

File: AllSamples/Samples/Samples_Source/peer networking/peeridentity viewer/cs/dlgnewidentity.cs
//=====================================================================
// File:      DlgNewIdentity.cs
//
// Summary:   Dialog to create a new PeerIdentity.
//
//---------------------------------------------------------------------
//  This file is part of the Microsoft .NET Framework SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//---------------------------------------------------------------------

using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Net.PeerToPeer;
using System.Windows.Forms;

namespace IdentityViewer
{
    /// <summary>
    /// Dialog to create a new PeerIdentity.
    /// </summary>
    public class DlgNewIdentity : System.Windows.Forms.Form
    {
        private PeerIdentity identityValue = null;

        private System.Windows.Forms.Label labelName;
        private System.Windows.Forms.TextBox textBoxName;
        private System.Windows.Forms.Label labelClassifier;
        private System.Windows.Forms.TextBox textBoxClassifier;
        private System.Windows.Forms.Button buttonOk;
        private System.Windows.Forms.Button buttonCancel;

        private System.ComponentModel.Container components = null;

        /// <summary>
        /// Constructor to create the dialog.
        /// </summary>
        public DlgNewIdentity()
        {
            InitializeComponent();

            // Default the name to the user's login name.
            textBoxName.Text = System.Environment.UserName;
        }

        /// <summary>
        /// Clean up any resources being used.
        /// </summary>
        protected override void Dispose( bool disposing )
        {
            if( disposing )
            {
                if(components != null)
                {
                    components.Dispose();
                }
            }
            base.Dispose( disposing );
        }

        #region Windows Form Designer generated code
        /// <summary>
        /// Required method for Designer support - do not modify
        /// the contents of this method with the code editor.
        /// </summary>
        private void InitializeComponent()
        {
            this.labelName = new System.Windows.Forms.Label();
            this.textBoxName = new System.Windows.Forms.TextBox();
            this.textBoxClassifier = new System.Windows.Forms.TextBox();
            this.labelClassifier = new System.Windows.Forms.Label();
            this.buttonOk = new System.Windows.Forms.Button();
            this.buttonCancel = new System.Windows.Forms.Button();
            this.SuspendLayout();

            // 
            // labelName
            // 
            this.labelName.Location = new System.Drawing.Point(16, 16);
            this.labelName.Name = "labelName";
            this.labelName.Size = new System.Drawing.Size(48, 16);
            this.labelName.TabIndex = 0;
            this.labelName.Text = "Name:";

            // 
            // textBoxName
            // 
            this.textBoxName.Anchor = ((System.Windows.Forms.AnchorStyles)(((System.Windows.Forms.AnchorStyles.Top | System.Windows.Forms.AnchorStyles.Left) | System.Windows.Forms.AnchorStyles.Right)));
            this.textBoxName.Location = new System.Drawing.Point(72, 16);
            this.textBoxName.Name = "textBoxName";
            this.textBoxName.Size = new System.Drawing.Size(204, 19);
            this.textBoxName.TabIndex = 1;

            // 
            // textBoxClassifier
            // 
            this.textBoxClassifier.Anchor = ((System.Windows.Forms.AnchorStyles)(((System.Windows.Forms.AnchorStyles.Top | System.Windows.Forms.AnchorStyles.Left) | System.Windows.Forms.AnchorStyles.Right)));
            this.textBoxClassifier.Location = new System.Drawing.Point(72, 56);
            this.textBoxClassifier.Name = "textBoxClassifier";
            this.textBoxClassifier.Size = new System.Drawing.Size(204, 19);
            this.textBoxClassifier.TabIndex = 3;

            // 
            // labelClassifier
            // 
            this.labelClassifier.Location = new System.Drawing.Point(16, 56);
            this.labelClassifier.Name = "labelClassifier";
            this.labelClassifier.Size = new System.Drawing.Size(56, 16);
            this.labelClassifier.TabIndex = 2;
            this.labelClassifier.Text = "Classifier:";

            // 
            // buttonOk
            // 
            this.buttonOk.Anchor = ((System.Windows.Forms.AnchorStyles)((System.Windows.Forms.AnchorStyles.Bottom | System.Windows.Forms.AnchorStyles.Right)));
            this.buttonOk.Location = new System.Drawing.Point(71, 96);
            this.buttonOk.Name = "buttonOk";
            this.buttonOk.TabIndex = 4;
            this.buttonOk.Text = "OK";
            this.buttonOk.Click += new System.EventHandler(this.buttonOk_Click);

            // 
            // buttonCancel
            // 
            this.buttonCancel.Anchor = ((System.Windows.Forms.AnchorStyles)((System.Windows.Forms.AnchorStyles.Bottom | System.Windows.Forms.AnchorStyles.Right)));
            this.buttonCancel.DialogResult = System.Windows.Forms.DialogResult.Cancel;
            this.buttonCancel.Location = new System.Drawing.Point(159, 96);
            this.buttonCancel.Name = "buttonCancel";
            this.buttonCancel.TabIndex = 5;
            this.buttonCancel.Text = "Cancel";
            this.buttonCancel.Click += new System.EventHandler(this.buttonCancel_Click);

            // 
            // DlgNewIdentity
            // 
            this.AcceptButton = this.buttonOk;
            this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
            this.CancelButton = this.buttonCancel;
            this.ClientSize = new System.Drawing.Size(292, 133);
            this.ControlBox = false;
            this.Controls.Add(this.buttonCancel);
            this.Controls.Add(this.buttonOk);
            this.Controls.Add(this.textBoxName);
            this.Controls.Add(this.labelName);
            this.Controls.Add(this.textBoxClassifier);
            this.Controls.Add(this.labelClassifier);
            this.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedDialog;
            this.Name = "DlgNewIdentity";
            this.StartPosition = System.Windows.Forms.FormStartPosition.CenterParent;
            this.Text = "Create New Identity";
            this.ResumeLayout(false);
            this.PerformLayout();
        }
        #endregion


        /// <summary>
        /// The PeerIdentity that was imported.
        /// </summary>
        /// <value></value>
        public PeerIdentity Identity
        {
            get
            {
                return identityValue;
            }
        }

        /// <summary>
        /// Validate the data in the dialog and create the Identity.
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        private void buttonOk_Click(object sender, System.EventArgs e)
        {
            if (CreateIdentity())
            {
                this.DialogResult = DialogResult.OK;
                Close();
            }
        }

        /// <summary>
        /// Try to create the identity.
        /// </summary>
        /// <returns></returns>
        private bool CreateIdentity()
        {
            // Validate the classifier
            textBoxClassifier.Text = textBoxClassifier.Text.Trim();
            if (textBoxClassifier.TextLength >= 150)
            {
                MessageBox.Show(this, "The classifier must be less than 150 characters",
                    "Invalid Classifier", MessageBoxButtons.OK, MessageBoxIcon.Error);
                return false;
            }

            // Validate the friendly name
            textBoxName.Text = textBoxName.Text.Trim();
            if (textBoxName.TextLength >= 256)
            {
                MessageBox.Show(this, "Identity friendly must be less than 256 characters",
                    "Invalid Name", MessageBoxButtons.OK, MessageBoxIcon.Error);
                return false;
            }

            try
            {
                identityValue = new PeerIdentity(textBoxName.Text, textBoxClassifier.Text);
                return true;
            }
            catch (Exception e)
            {
                Utilities.DisplayException(e, this);
                return false;
            }
        }

        /// <summary>
        /// Close the dialog without creating a new identity.
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        private void buttonCancel_Click(object sender, System.EventArgs e)
        {
            Close();
        }
    }
}
